app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsRTDB.kt

```diff
@@ -60,4 +69,21 @@ class UserCardsRTDB {
         }
         return future
     }
+
+    /**
+     * Retrieves all the cards asynchronously from the database
+     */
+    fun getAllCardsFromCollection(): CompletableFuture<DataSnapshot> {
+        val future = CompletableFuture<DataSnapshot>()
+        userCardCollection?.get()?.addOnSuccessListener {
+            if (it.value == null) {
+                future.completeExceptionally(NoSuchFieldException("you don't have any card in your collection"))

```

Comment:
`NoSuchFieldException` is intended for Java's reflection stack, i.e., a class doesn't have the requested field. `NoSuchElementException` would be a better fit.